### PR TITLE
Add support for XPU device in `PrefetchLoader`

### PR DESCRIPTION
**Pull request by DamianSzwichtenberg**
Device abstraction is now covered by an internal class, called `PrefetchLoaderDevice`. It handles the device selection mechanism, stream creation, and synchronization.

### CODE REVIEW

File: torch_geometric/typing.py [hunk 1]
```diff
@@ -205,6 +205,17 @@ def masked_select_nnz(src: SparseTensor, mask: Tensor,
...
 
 
+try:
+    import intel_extension_for_pytorch  # noqa
+    WITH_IPEX = True
+except (ImportError, OSError) as e:
+    if isinstance(e, OSError):
+        warnings.warn("An issue occurred while importing"
+                      "'intel-extension-for-pytorch'. "
+                      f"Disabling its usage. Stacktrace: {e}")
```
Comment (rusty1s) [inline torch_geometric/typing.py:215]:
```suggestion
```

File: torch_geometric/loader/prefetch.py [hunk 2]
```diff
@@ -15,51 +17,91 @@ class PrefetchLoader:
...
             return batch
         if isinstance(batch, (list, tuple)):
             return [self.non_blocking_transfer(v) for v in batch]
         if isinstance(batch, dict):
             return {k: self.non_blocking_transfer(v) for k, v in batch.items()}
 
-        batch = batch.pin_memory()
-        return batch.to(self.device, non_blocking=True)
+        device = self.device_mgr.get_device()
+        batch = batch.pin_memory(device)
```
Comment (rusty1s) [inline torch_geometric/loader/prefetch.py:87]: Can you clarify? `pin_memory` doesn't seem to have a `device` argument. https://pytorch.org/docs/stable/generated/torch.Tensor.pin_memory.html

File: torch_geometric/loader/prefetch.py [hunk 3]
```diff
@@ -15,51 +17,91 @@ class PrefetchLoader:
...
-
         self.loader = loader
-        self.device = torch.device(device)
-
-        self.is_cuda = torch.cuda.is_available() and self.device.type == 'cuda'
+        self.device_mgr = self.PrefetchLoaderDevice(device)
 
     def non_blocking_transfer(self, batch: Any) -> Any:
-        if not self.is_cuda:
+        if not self.device_mgr.is_gpu:
```
Comment (rusty1s) [inline torch_geometric/loader/prefetch.py:79]: What does `mgr` stand for?

File: torch_geometric/loader/prefetch.py (hunk 2 above)
Comment (DamianSzwichtenberg) [reply torch_geometric/loader/prefetch.py:87]: I think this doc is not correct, take a look at the below piece of `native_functions.yaml` ([source](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L4330))
```
- func: pin_memory(Tensor(a) self, Device? device=None) -> Tensor(a)
  variants: method
```
The default device is CUDA, others have to be specified explicitly.

File: torch_geometric/loader/prefetch.py (hunk 3 above)
Comment (DamianSzwichtenberg) [reply torch_geometric/loader/prefetch.py:79]: `mgr` is an abbreviation for `manager`.

**Review by rusty1s — APPROVED** [review]

**Pull request merged**